SetupTearDown.ps1

```diff
@@ -118,16 +118,11 @@ param(
         Write-Host "`nSSHD default config files already exist"
     }
 
-    if (-NOT (Test-Path "$env:ProgramData\ssh\administrators_authorized_keys"))
-    {
+    if (-NOT (Get-Content -Path "$env:ProgramData\ssh\administrators_authorized_keys" | ForEach-Object{$_ -match $authorizedKey})) {

```

Comment:
I don't think this will work the way you want it to. If Get-Content returns more than one line, the result of the Get-Content | Foreach {} is going to be an array and an array is always considered "true" when interpreted as bool. So, this line will work if the authorized_keys file is empty or has one line, but will always skip adding the key if there is more than one key in the file, even if the key being checked isn't in there.

You could do something like this:
$isPresent = $false
PS C:\Users\marsam.REDMOND> Get-Content <key file> | foreach { $isPresent = $isPresent -OR $_ -match $authorizedKey }
if (-NOT $IsPresent)
{...}

Reply:
modified the statement so that it just checks the string!